packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -5,14 +5,43 @@
 import 'basic.dart';
 import 'framework.dart';
 
-/// A place in the overlay that can contain a widget.
+/// A place in an [Overlay] that can contain a widget.
+///
+/// Overlay entires are inserted into an [Overlay] using the
+/// [OverlayState.insert] or [OverlayState.insertAll] functions. To find the
+/// closest enclosing overlay for a given [BuildContext], use the [Overlay.of]
+/// function.
+///
+/// An overlay entry can be in at most one overlay at a time. To remove an entry
+/// from its overlay, call the [remove] function on the overlay entry.
+///
+/// For example, [Draggable] uses an [OverlayEntry] to show the drag avatar that
+/// follows the user's finger across the screen after the drag begins. Using the
+/// overlay to display the drag avatar lets the avatar float over the other
+/// widgets in the app. When the drag is over, [Draggable] removes the entry
+/// from the overlay to remove the drag avatar from view.
+///
+/// See also:
+///
+///  * [Overlay]
+///  * [OverlayState]
+///  * [WidgetsApp]
+///  * [MaterialApp]
```

Comment:
Suggest mentioning Stack here, since Overlays are kind of a syntactic sugar on top of Stack.